core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/dataset/DatasetResource.scala

```diff
@@ -1054,81 +1056,95 @@ class DatasetResource {
   }
 
   /**
-    * Retrieves a ZIP file for a specific dataset version.
+    * Retrieves a ZIP file for a specific dataset version or the latest version.
     *
     * @param pathStr The dataset version path in the format: /ownerEmail/datasetName/versionName
     *                Example: /user@example.com/dataset/v1
-    *
-    * @param user the session user.
+    * @param getLatest When true, retrieves the latest version regardless of the provided path.
+    * @param did The dataset ID (used when getLatest is true).
+    * @param user The session user.
     * @return A Response containing the dataset version as a ZIP file.
     */
   @GET
   @Path("/version-zip")
   def retrieveDatasetVersionZip(
       @QueryParam("path") pathStr: String,
+      @QueryParam("getLatest") getLatest: Boolean,
+      @QueryParam("did") did: UInteger,
       @Auth user: SessionUser
   ): Response = {
-    val uid = user.getUid
-    val decodedPathStr = URLDecoder.decode(pathStr, StandardCharsets.UTF_8.name())
-    val (_, dataset, dsVersion, _) =
-      resolvePath(Paths.get(decodedPathStr), shouldContainFile = false)
-
-    withTransaction(context) { ctx =>
-      val did = dataset.getDid
-      val dvid = dsVersion.getDvid
+    val (dataset, version) = if (getLatest) {
+      getLatestVersionInfo(did, user)
+    } else {
+      resolveAndValidatePath(pathStr, user)
+    }
+    val targetDatasetPath = PathUtils.getDatasetPath(dataset.getDid)
+    val fileNodes = GitVersionControlLocalFileStorage.retrieveRootFileNodesOfVersion(
+      targetDatasetPath,
+      version.getVersionHash
+    )
 
-      if (!userHasReadAccess(ctx, did, uid)) {
-        throw new ForbiddenException(ERR_USER_HAS_NO_ACCESS_TO_DATASET_MESSAGE)
+    val streamingOutput = new StreamingOutput {
+      override def write(outputStream: OutputStream): Unit = {
+        Using(new ZipOutputStream(outputStream)) { zipOutputStream =>
+          fileNodes.asScala.foreach { fileNode =>
+            val zipEntryName = fileNode.getRelativePath.toString
+            val filePath = fileNode.getAbsolutePath
```

Comment:
file node can have children, the children is not iterated. So any files under the directory will not be downloaded. Can you add logic to check if this fileNode is directory and iterate its children? Also if a fileNode is a directory, it should not be written as a file. Currently the directory is in the form of empty file in the zip

Reply:
I can fix this issue, but I just want to ask why is this issue not handled in the previous [PR](https://github.com/Texera/texera/pull/2820)?

Reply:
This request should be done in a separate PR as a "bug fix". Addressing it here would produce a potential bug. It should

Reply:
Sure.